### PR TITLE
A plugable PoorManProfiler singleton

### PR DESCRIPTION
**Pull request by drrtuy**

### CODE REVIEW

File: primitives/primproc/primitiveserver.h [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 #include "service.h"
 
-//#define PRIMPROC_STOPWATCH
+#define PRIMPROC_STOPWATCH
```
Comment (mariadb-AlexeyAntipovsky) [inline primitives/primproc/primitiveserver.h]: Is it really needed to be set in production?

File: utils/loggingcpp/poormanprofiler.inc [hunk 2]
```diff
@@ -0,0 +1,140 @@
+#ifndef POORMANPROFILER_
+#define POORMANPROFILER
```
Comment (mariadb-AlexeyAntipovsky) [inline utils/loggingcpp/poormanprofiler.inc]: Guard name mismatch

File: utils/loggingcpp/poormanprofiler.inc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            stopwatch = stopwatchMapIter->second;
+
+            // If any threads have been active in the last 5 seconds, kick out.
+            if (((tvCurrent.tv_sec - stopwatch->fTvLast.tv_sec) < 5) || stopwatch->isActive())
+            {
+                procIsIdle = false;
+                break;
+            }
+
+            stopwatchMapIter++;
```
Comment (mariadb-AlexeyAntipovsky) [inline utils/loggingcpp/poormanprofiler.inc:74]: I'm afraid there may be race condition here (i.e. in the case when inserting into the unordered_map from another thread will cause rehashing)

File: utils/loggingcpp/poormanprofiler.inc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        if (err)
+            std::cerr << "Error creating thread to complete Stopwatches." << endl;
+
+        stopwatchThreadCreated = true;
+    }
+}
+
+inline logging::StopWatch* PoorManProfiler::start()
+{
+    auto stopwatchMapIter = stopwatchMap.find(pthread_self());
```
Comment (mariadb-AlexeyAntipovsky) [inline utils/loggingcpp/poormanprofiler.inc:117]: And here too...

**Review by mariadb-AlexeyAntipovsky — CHANGES_REQUESTED** [review]

File: primitives/primproc/primitiveserver.h (hunk 1 above)
Comment (drrtuy) [reply primitives/primproc/primitiveserver.h]: Surely not.

**Review by mariadb-AlexeyAntipovsky — APPROVED** [review]

**Pull request merged**